### DISCUSSION TITLE
Should we use SEND_PWM_BY_TIMER?

### DISCUSSION THREAD

**Discussion by chino** [Arduino-IRremote/Arduino-IRremote#1059] [Q&A] [answered]
I'm not sure if it's better to enable SEND_PWM_BY_TIMER or what the tradeoff is?
Thanks!

**Comment by ArminJo** [comment, marked as answer]
Good question 👍 
I extended the readme.
`SEND_PWM_BY_TIMER`: Disables carrier PWM generation in software and use hardware PWM (by timer). Has the advantage of more exact PWM generation, especially the duty cycle (which is not very relevant for most IR receiver circuits), and the disadvantage of using a hardware timer, which in turn is not available for other libraries and to fix the send pin (but not the receive pin) at the [dedicated timer output pin(s)](https://github.com/Arduino-IRremote/Arduino-IRremote#timer-and-pin-usage). Is enabled for ESP32 and RP2040 in all examples, since they support PWM gereration for each pin without using a shared resource (timer).

**Comment by chino** [comment]
So real world applications don't really need that amount of precision?

**Reply by ArminJo** [reply]
Didn't I write this?

**Reply by chino** [reply]
Sorry I guess I was trying to confirm what kind of applications might not need it.  Thank you for clearing this up.

**Reply by ArminJo** [reply]
No need to say sorry, If I am not clear enough, it is definitely my fault!
Should I write better "which is not very relevant for real world IR receiver circuits" or "which is not very relevant for most IR receiver circuits" ?
Please be frankly!

**Reply by chino** [reply]
https://github.com/Arduino-IRremote/Arduino-IRremote#hardware-pwm-signal-generation-for-sending

That part there says, 
```
Since each hardware timer has its dedicated output pins, you must change timer to change PWM output.
```

Are you saying that generally the default hardware timer is tied to a specific pin?  Does it report which timer it wants to use and which pin we should use for the current hardware?

**Reply by chino** [reply]
Also it says, `The same timer as for the receiver is used` which makes me think that once I enable `SEND_PWM_BY_TIMER` that I can only use a single pin for both send/receive.

Hence the following code is most likely broken (using a Arduino Duemilanove) because I'm using different pins whereas the hardware timer (for both) is only tied to a single pin?
```
#define IR_RECEIVE_PIN 2 
#define IR_SEND_PIN 3 
#define SEND_PWM_BY_TIMER
```
